src/services/factories/getPriceEstimation.ts

```diff
@@ -0,0 +1,73 @@
+import BigNumber from 'bignumber.js'
+import { assert } from '@gnosis.pm/dex-js'
+
+import { ZERO_BIG_NUMBER } from 'const'
+
+import { TheGraphApi } from 'api/thegraph/TheGraphApi'
+import { TokenList } from 'api/tokenList/TokenListApi'
+import { Network } from 'types'
+
+export interface GetPriceParams {
+  numeratorTokenId: number
+  denominatorTokenId: number
+}
+
+export function getPriceEstimationFactory(factoryParams: {
+  theGraphApi: TheGraphApi
+  tokenListApi: TokenList
+}): (params: GetPriceParams) => Promise<BigNumber> {
+  const { theGraphApi, tokenListApi } = factoryParams
+
+  // Only make sense to fetch prices for mainnet, thus we won't accept networkId parameter on this service
+  const networkId = Network.Mainnet
+
+  const tokenIdsSet = new Set(tokenListApi.getTokens(networkId).map(token => token.id))
+  const tokenIds = Array.from(tokenIdsSet).sort()
+
+  /**
+   * Given a tokenId, checked whether it's already in our list.
+   * Returns true if tokenId was added to the list and it requires sorting
+   */
+  function addTokenId(tokenId: number): boolean {
+    if (!tokenIdsSet.has(tokenId)) {
+      tokenIdsSet.add(tokenId)
+
+      tokenIds.push(tokenId)
+
+      return true
+    }
+
+    return false
+  }
+
+  return async (params: GetPriceParams): Promise<BigNumber> => {
+    const { numeratorTokenId, denominatorTokenId } = params
+
+    assert(numeratorTokenId !== denominatorTokenId, 'Token ids cannot be the equal')
+
+    // As much as it'd be visually pleasing, cannot simplify it to a single expression
+    // We need to execute both functions.
+    let needsToSort = addTokenId(numeratorTokenId)
```

Comment:
> As much as it'd be visually pleasing, cannot simplify it to a single expression. We need to execute both functions

I didn't fully get it :) 
I guess you mean this is not possible for some reason:
`const needsToSort = addTokenId(numeratorTokenId) || addTokenId(denominatorTokenId)`

Reply:
Yes!
@Velenir proved me wrong right away of course, but I rather not do that.
I'll update the comment to be clearer.

Reply:
Done, no more implicit or needing comments.